File: iris-did/src/main/java/city/smartb/iris/did/model/ControledJsonLdObject.java
package city.smartb.iris.did.model;

import city.smartb.iris.jsonld.JsonLdObject;
import city.smartb.iris.jsonld.reader.JsonFieldReader;

import java.util.Map;

public abstract class ControledJsonLdObject extends JsonLdObject {

	public static final String JSON_LD_CONTROLER = "controller";

	public ControledJsonLdObject(Map<String, Object> jsonLdObject) {
		super(jsonLdObject);
	}

	public ControledJsonLdObject(Map<String, Object> jsonLdObject, JsonFieldReader fieldReader) {
		super(jsonLdObject, fieldReader);
	}

	public String getController() {
		return this.get(JSON_LD_CONTROLER).asString();
	}

}

File: iris-ld/iris-did/src/main/kotlin/city/smartb/iris/ld/did/ControledJsonLdObject.kt
package city.smartb.iris.ld.did

import city.smartb.iris.ld.jsonld.JsonLdObject
import city.smartb.iris.ld.jsonld.reader.JsonFieldReader

abstract class ControledJsonLdObject : JsonLdObject {
    constructor(jsonLdObject: Map<String, Any>) : super(jsonLdObject) {}
    constructor(jsonLdObject: Map<String, Any>, fieldReader: JsonFieldReader) : super(jsonLdObject, fieldReader) {}

    companion object {
        const val JSON_LD_CONTROLER = "controller"
    }

    val controller: String?
        get() = this[JSON_LD_CONTROLER].asString()
}
